subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -3598,7 +3626,10 @@ static int handle_request(struct coap_packet *request,
 				/* remove observer */
 				r = engine_remove_observer(token, tkl);
 				if (r < 0) {
-					LOG_ERR("remove observe error: %d", r);
+					r = engine_remove_observer_by_path(&msg->path);
```

Comment:
I would argue that according to the spec the outer call to
`r = engine_remove_observer(token, tkl);` should be removed. But I guess it is less invasive to do it like this

Reply:
After some discussions with various ppl, it seems token matching is likely implicitly assumed in the spec. The rigorous interpretation of the spec would actually be to enforce token AND path matching. I think this is good for now.